pkg/odo/cli/component/create.go

```diff
@@ -319,7 +319,11 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 					supDevfileCatalogList = append(supDevfileCatalogList, devfileComponent)
 				}
 			}
-			componentType = ui.SelectDevfileComponentType(supDevfileCatalogList)
+
+			// devfile.yaml is not present, user has to specify the component type
+			if !util.CheckPathExists(DevfilePath) {
```

Comment:
can we move this check before listing from registry, as for the case devfile is present it might save one call.

Reply:
Nice suggestion! I can do to improve the performance.